### PR TITLE
Add JSDoc and perform minor cleanup

### PR DESCRIPTION
**Pull request by sandersky**
@kellyselden here is the first smaller PR for my behemoth PR. This one doesn't change anything functionally as it is just adding JSDoc and performing some minor cleanup.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,10 +8,6 @@ node_js:
 
 sudo: false
 
-cache:
```
Comment (kellyselden) [inline .travis.yml]: caching node_modules is good for testing times. why remove?

Comment (sandersky) [inline .travis.yml]: I've ran into many issues in the past with caching node_modules with Travis where it ends up getting cached in a bad state. When this happens it is not obvious that clearing the cache will resolve issues and many hours end up being wasted digging into the issue.

Comment (rwjblue) [inline .travis.yml]: @sandersky - I have also felt this pain and definitely understand the goal/intent here, but I think this addon should try to stick as close to the default addon blueprint as possible.  Straying from that makes future upgrades harder...

File: tests/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (rwjblue) [inline tests/.eslintrc]: Can you rename this to `.eslintrc.js` and `module.export` this?  Using JS files will allow us to cascade rules a bit easier...

**Comment by rwjblue** [comment]
I left two minor comments inline, but other than those this looks great!

**Comment by sandersky** [comment]
Thanks @rwjblue, I have addressed your concerns.

**Pull request closed without merging**